scripts/collect_miss_cnt.py

```
@@ -46,46 +106,374 @@ def miss_cnt_from_log(path: Path) -> str:
     return matches[-1] if matches else "-"
 
 
-def main(base_dir: str = "css-gen-op"):
-    base = Path(base_dir).resolve()
-    rows: list[tuple[str, ...]] = []
+def collect_rows(base: Path) -> List[ReportRow]:
+    """Gather per-site statistics from the trace folders."""
+
+    if not base.exists():
+        raise SystemExit(f"Base directory {base} does not exist")
```

Ruff (0.14.8)

Avoid specifying long messages outside the exception class

(TRY003)